src/gc/impl/conservative/gc.d

```diff
@@ -2575,9 +2604,54 @@ struct Gcx
         begin = start = currTime;
 
         debug(COLLECT_PRINTF) printf("Gcx.fullcollect()\n");
+        bool doFork = config.fork;
+        version (COLLECT_PARALLEL)
+            bool doParallel = config.parallel > 0;
+        else
+            enum doParallel = false;
+
+        assert(!(doFork && doParallel), "No reason in mixing threads and fork");
         //printf("\tpool address range = %p .. %p\n", minAddr, maxAddr);
 
+        // If there is a mark process running, check if it already finished.
+        // If that is the case, we move to the sweep phase.
+        // If it's still running, either we block until the mark phase is
+        // done (and then sweep to finish the collection), or in case of error
+        // we redo the mark phase without forking.
+        if (collectInProgress && doFork)
```

Comment:
As discussed move this section to a function if possible.

Reply:
Done, see commit 
https://github.com/dlang/druntime/pull/2604/commits/d3fe6ad574d09c8facc70172f9162b1d3c256fc6